MQG goes WindowsForms

CToellerPriv opened the pull request:
Projekt für WindowsFormsAnwendung ergänzt. Dazu waren einige Änderungen im Ausgangsprojekt nötig. Funktion ist für Kommandozeile und WindowsForms grob getestet. Code ist noch nicht gesäubert. Problem bei Komandozeile/Ausgangsprojekt: Setzen der Pfadvariablen in Program. Mit optionalem Parameter gelöst. Wäre das auch was für die anderen Parameter beim Suchen?

CToellerPriv committed:
Projekt für WindowsFormsAnwendung ergänzt. Dazu waren neinige Änderunegn im Ausgangsprojekt nötig. Funktion ist für Kommandozeile und WindowsForms grob getestet. Code ist noch nicht gesäubert. Problem bei Komandozeile/Ausgangsprojekt: Setzen der Pfadvariablen in Program. Mit optionalem Parameter gelöst. Wäre das auch was für die anderen Parameter beim Suchen?

CToellerPriv committed:
In MQGGui gibt es eine Vorschau mit bearbeiten Funktion. Texte und Richtig/Falsch können geändert werden und werden gespeichert.
Hinzufügen und Löschen wäre der nächste Schritt.
Alles davon ist jeweils auf 5Antworten ausgelegt!

Toeller committed:
Überarbeitung Namensgebung der Datei.

Toeller committed:
Differenzen gemerged

Toeller committed:
Die Tests liefen nichtmehr. Es waren Umfangreiche Updates des Testframworks nötig.

Toeller committed:
Test hat den richtigen Namen, funktioniert und liefert ein korrektes Ergebnis.
Das Setzen der Kategorie wurde geändert und nun auch im Test angepasst.

Toeller committed:
kleinigkeiten

Toeller committed:
Begin Umbau für "Neu".
Achtung noch mit Fehlern!

Toeller committed:
Ein paar Fehler behoben ;-)

Toeller committed:
Anfang XML-Import mit Fehlern.

CToellerPriv committed:
Öffnen von XML funktioniert. Auch mit Antworten <5.

CToellerPriv committed:
csv und xml funktionieren mit öffnen auch während einer Sitzung.

CToellerPriv commented:
Funktion ist da. Code ist unsauber. Viele Dopplungen in ViewMain...
@klasenhhbk : Machst Du die PeerReview?

CToellerPriv committed:
Neu und Löschen funktioniert auch!
Bei neuem Import wird die vorhandene XML-Datei überschrieben!

CToellerPriv commented:
@klasenhhbk: Soll ich mal mergen oder traust Du Dich ;-)

Toeller committed:
Mit Installer Projekt (Erweiterung:Microsoft Visual Installer Projects 2022 -> Setup-Projekt).

klasenhhbk commented:
Du kannst du gerne weitermachen und alles mergen. @CToellerPriv 

Ich bin wahrscheinlich erst ab dem 1.4. wieder in der Lage in das Projekt hier zu schauen. Gerade hustle ich mit einer UB-Vorbereitung.

Toeller committed:
Squirrel für Updates in die Projekte eingebunden.

Toeller committed:
Squirrel wieder deinstalliert, da DeltaCompressionDotNet in aktueller Version nicht unterstützt wird.
Vielleicht: https://github.com/velopack/velopack?tab=readme-ov-file ?
Oder https://github.com/ravibpatel/AutoUpdater.NET ?

Toeller committed:
upss...

Toeller committed:
Lernmschine ergänzt

Toeller committed:
Index geändert, damit nach dem speichern nicht wieder von vorne begonnen werden muss.

CToellerPriv committed:
Lernmschinen Dateien hinzugefügt.

Toeller committed:
Ergäünzung der View um UCs für Unterrichtsfach, Thema und Fach/Lernstufe
Auswahl der Unterrichtsfächer unf Themen mit Comboboxen beim neuanlegen.
ModelXML begonnen.

Toeller committed:
Neu, Ändern und Lernen funktionieren. Hin und wieder treten noch Reihenfolgen fehler auf?!?

Toeller committed:
Löschen ist nun auch integriert.
Bei Beenden vom Lernen wird aber auf die erste Karte im gesammten Set zurückgesprungen und nicht auf die erste Karte im Lernset. Außerdem können auch Themen bei Fächern angezeigt werden, die es dort garnicht gibt.
